### ISSUE TITLE
The tokenizer is slow when the input string is long.

### ISSUE THREAD

**Issue by HaukurPall** [mideind/Tokenizer#40]
When the input into tokenizer is very long, the time it takes to tokenise the string into sentence does not grow linearly with the input length as one might expect.

Test case to demonstrate the issue:
```
from time import time

import tokenizer
from tqdm import tqdm


def create_test_sentence(num_words, num_sentences):
    """Create a test string which contains num_words words and num_sentences sentences."""
    sentences = []
    for i in range(num_sentences):
        sentence = ["Hæ"] * num_words
        sentences.append(" ".join(sentence))
        sentences[-1] += "."
    return " ".join(sentences) + "\n"


def run_multi_sentence_test(num_words, num_sentences, num_lines):
    print(f"Running test with {num_words} words and {num_sentences} sentences in each line and {num_lines} lines.")
    start = time()
    test_data = list(create_test_sentence(num_words, num_sentences) for _ in range(num_lines))
    count = 0
    for line in tqdm(test_data):
        for sent in tokenizer.split_into_sentences(line, original=True):
            count += 1
    end = time()
    assert count == num_lines * num_sentences
    print("number of sentences encountered:", count)
    print("time taken:", end - start)
    print(f"{(end - start) / count:.6f} seconds per sentence")


num_words = 20
num_sentences = 40
num_lines = 300
run_multi_sentence_test(num_words, num_sentences, num_lines)
# Running test with 20 words and 40 sentences in each line and 300 lines.
# ...
# number of sentences encountered: 12000
# time taken: 17.760034322738647
# 0.001480 seconds per sentence

num_sentences = 1
num_lines = 300 * 40
run_multi_sentence_test(num_words, num_sentences, num_lines)
# Running test with 20 words and 1 sentences in each line and 12000 lines.
# ...
# number of sentences encountered: 12000
# time taken: 3.183797597885132
# 0.000265 seconds per sentence

num_sentences = 300 * 40
num_lines = 1
run_multi_sentence_test(num_words, num_sentences, num_lines)
# Running test with 20 words and 12000 sentences in each line and 1 lines.
# Does not return a value in 5 minutes.
```

**Comment by HaukurPall** [comment]
I ran a profiler on these test cases and the problem is in Tok.split(), see https://github.com/mideind/Tokenizer/blob/9cd18923b2a99776136ab8469ebf4bbaa1f2b91f/src/tokenizer/tokenizer.py#L208

I will submit a PR to fix this issue

**Comment by HaukurPall** [comment]
Fixed in #41
